test/controllers/matches.js

```diff
@@ -0,0 +1,196 @@
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            response.body.should.be.a('array');
+            response.body.should.have.lengthOf(1);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should fail because of not being authenticated', (done) => {
+      chai.request(server)
+        .get('/games/2/match')
+        .send()
+        .catch((err) => {
+          err.response.should.be.json;
+          err.response.body.should.have.property('error');
+          err.response.body.error.should.equal('User is not logged in');
+          err.should.have.status(401);
+        })
+        .then((response) => {
+          done();
+        });
+    });
+  });
+  describe('/users/:userId/matches/emailSummary GET', () => {
+    it('should be successful emailing the list of matches by user', (done) => {
+      const sendUserHistory = sinon.spy(matchesMailer, 'sendUserHistory');
+      successfulLogin().then((res) => {
+        chai.request(server)
+          .get('/users/1/matches/emailSummary')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            sendUserHistory.restore();
+            sinon.assert.calledOnce(sendUserHistory);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should be successful emailing the requested user history without being admin', (done) => {
+      const sendUserHistory = sinon.spy(matchesMailer, 'sendUserHistory');
+      successfulLoginNotAdmin().then((res) => {
+        chai.request(server)
+          .get('/users/2/matches/emailSummary')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            sendUserHistory.restore();
+            sinon.assert.calledOnce(sendUserHistory);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should fail because user is not admin and is requesting for anothed user_id', (done) => {
+      successfulLoginNotAdmin().then((res) => {
+        chai.request(server)
+          .get('/users/1/matches/emailSummary')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .catch((err) => {
+            err.response.should.be.json;
+            err.response.body.error.should.equal('User is not allowed');
+            err.response.body.should.have.property('error');
```

Comment:
I would add here another spy to check that the `matchesMailer.sendUserHistory()` wasn't called.